### PR TITLE
1.16.5 to 1.17 update

### PR DESCRIPTION
**Pull request by Dragon-Seeker**
I'm targeting the main 1.16 Dev branch for now but A separate branch should be created for this PR.

This PR contains a set changes to get tropicraft Forge working on 1.17 Minecraft. Currently listed are the changes made the directly impacted the code base:

- Models
  - Due to the new Entity Model format introduced into 1.17, all the models needed to be converted to this new system.
    -  Some models are minorly broken due to bad copying of code meaning some may have incorrect position values, wrong model block size or a second head specifically on the White Lipped Peccary (I really have no clue how this has happened)
  - General entity rendering has been pushed to a subscriber event and so has the new model layers.
    - This did impact issues with getting the scuba gear model's due to how the original code was storing the models so a work around was put into place
  - Speaking of armor, I had to change how it called for a model by adding a create model method within the PlayerHeadPieceModel and the ScubaModel
    - Theses are not working as of the moment as it will show the texture incorrectly or like the ashen mask, it will show a red box around the players head
---------------------------------------------------------------------------------------
- Mapping
  - The mapping has been updated to use mojmap as MCP is no longer support(RIP), but a combination of mojmap and Parchment should be fine replacment
  - Forge has also changed the naming convetion of tile entity's to block entitys so with the future of the PR, such should be reflected on all class files and folders 
---------------------------------------------------------------------------------------
- Dimension
  - Update to how mangrove's work with the removal of methods that were put into the decorator's
  - Change over to Context based methods for a lot of the placers I.E the foliage placer and feature's
  - MAJOR ISSUE: Due to some change between 1.16 to 1.17, access chucks outside the max range of 3x3 for a feature will crash the game so the Home feature should be changed or updated to a structure

Suggestion: I'd like to pr a change that I made within the fabric version of tropicraft the allows for leaf decay on the bigger trees. Its still experimental but would make farming and using tropicraft trees a lot easier. It's not included within this one as Ill do a separate branch within the future.

There might be other stuff missed but for more info on what might needed to be changed: [Link](https://gist.github.com/50ap5ud5/beebcf056cbdd3c922cc8993689428f4)

### CODE REVIEW

File: src/generated/test/data/tropicraft/worldgen/configured_feature/huge_golden_leather_fern.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
```
Comment (Gegy) [inline src/generated/test/data/tropicraft/worldgen/configured_feature/huge_golden_leather_fern.json]: Are these generated/test files being generated intentionally?

Comment (Dragon-Seeker) [reply src/generated/test/data/tropicraft/worldgen/configured_feature/huge_golden_leather_fern.json]: I think I did such back during debug step of attempting to fix some worldgen issues and tag problems. I'll remove it in the latest commit

**Comment by Cojomax99** [comment]
Looks like this branch could use a rebase!

**Comment by Dragon-Seeker** [comment]
> Looks like this branch could use a rebase!

That would require a 1.17 branch to target such, right?

**Comment by Dragon-Seeker** [comment]
Thanks github. If you rename a branch it will instead of renaming the branch, it will create a new branch with said branch as a base and rename that one meaning it will then delete the "old" one.

**Comment by Gegy** [comment]
Closing this for now as this has been merged into the 1.17 & 1.18 branches, thank you so much for the insane amount of work you've put into this PR, it has been such a huge help! 💜

**Pull request closed without merging**